### PR TITLE
OPENDINGUX: Force 320x240 for the rg99 and simplify build scripts

### PR DESCRIPTION
**Pull request by citral23**
The true resolution of the screen is 320x480, gui has the wrong aspect without requesting 320x240

Cleanup of build script since rs90 and rg99 use the same build/opk

Signed-off-by: Christophe Branchereau <cbranchereau@gmail.com>

### CODE REVIEW

File: backends/platform/sdl/opendingux/opendingux.mk [hunk 1]
```diff
@@ -1,7 +1,7 @@
 OD_EXE_STRIPPED := scummvm_stripped$(EXEEXT)
 bundle = od-opk
-ifeq ($(OPENDINGUX_TARGET), rg99)
-OPKNAME = rg99_rs90
+ifeq ($(OPENDINGUX_TARGET), rs90)
+OPKNAME = rs90_rg99
 else
```
Comment (phcoder) [inline backends/platform/sdl/opendingux/opendingux.mk:5]: I think we had a discussion to change it to rs90 but I don’t really care

**Review by phcoder — COMMENTED** [review]
Looks good. Just one comment and can you split it in 2-3 CLs. SDL change is separate from build script changes. It’s fine to keep them in the same PR, just separate CLs

**Comment by citral23** [comment]
Hi @phcoder, following out conversation I verified and the rs90 doesn't need env SDL_VIDEO_KMSDRM_SCALING_MODE=1, weither it's used or not it runs full screen either way. 

We can thus remove it and use the same .desktop file for both the rs90 and rg99 which simplifies the build and possible future maintenance.

Splitted the commits in 2 parts, you're right about that

KR

File: backends/platform/sdl/opendingux/opendingux.mk (hunk 1 above)
Comment (lephilousophe) [reply backends/platform/sdl/opendingux/opendingux.mk:5]: Yes, please avoid this special case for the buildbot.
We will add a note on packages list to inform that it covers rg99 as well.

**Comment by phcoder** [comment]
I believe now you forget to create default.rg99.desktop in the opk file

**Comment by citral23** [comment]
> I believe now you forget to create default.rg99.desktop in the opk file

No because it uses startUI.rs90.desktop, there's no need to duplicate them

File: backends/platform/sdl/opendingux/opendingux.mk (hunk 1 above)
Comment (citral23) [reply backends/platform/sdl/opendingux/opendingux.mk:5]: Ok

**Comment by phcoder** [comment]
> > I believe now you forget to create default.rg99.desktop in the opk file
> 
> No because it uses startUI.rs90.desktop, there's no need to duplicate them

So you mean gmenu2x will use default.rs90.desktop on rg99 if no default.rg99.desktop is present? Can you double-check by removing od-opk directory?

**Comment by citral23** [comment]
> > > I believe now you forget to create default.rg99.desktop in the opk file
> > 
> > 
> > No because it uses startUI.rs90.desktop, there's no need to duplicate them
> 
> So you mean gmenu2x will use default.rs90.desktop on rg99 if no default.rg99.desktop is present? Can you double-check by removing od-opk directory?

Yes, rs90 only uses rs90.desktop. RG99 uses rg99.desktop if present, rs90.desktop otherwise

File: backends/platform/sdl/opendingux/opendingux.mk (hunk 1 above)
Comment (phcoder) [reply backends/platform/sdl/opendingux/opendingux.mk:5]: Do we still need this if? Looks like second case covers everything

**Comment by phcoder** [comment]
Left one comment otherwise LGTM

File: backends/platform/sdl/opendingux/opendingux.mk (hunk 1 above)
Comment (lephilousophe) [reply backends/platform/sdl/opendingux/opendingux.mk:5]: Yes that's what I meant by removing the special case :D

Comment (citral23) [reply backends/platform/sdl/opendingux/opendingux.mk]: Yes, indeed, done.

**Review by phcoder — APPROVED** [review]

File: backends/graphics/surfacesdl/surfacesdl-graphics.cpp [hunk 2]
```diff
@@ -738,6 +738,7 @@ void SurfaceSdlGraphicsManager::getDefaultResolution(uint &w, uint &h) {
 	SDL_Rect const* const*availableModes = SDL_ListModes(&p, SDL_FULLSCREEN|SDL_HWSURFACE);
 	w = availableModes[0]->w;
 	h = availableModes[0]->h;
+	if (h > w) h /= 2;
```
Comment (ccawley2011) [inline backends/graphics/surfacesdl/surfacesdl-graphics.cpp:741]: Could you add a comment here?

Alternatively, does _window-> getDesktopResolution() give expected results for both RS90 and RG99?

Comment (citral23) [reply backends/graphics/surfacesdl/surfacesdl-graphics.cpp]: @ccawley2011 comment added. As the maintainer of the OD port I feel this is getting too close to the freeze to experiment alternatives now, I'd rather have this PR expedited to unbreak the rg99. Then we can have a discussion about how to handle things better for the next one

**Pull request merged**